### PR TITLE
Shopcart UI

### PR DESCRIPTION
**Pull request by thoughts1053**
This adds a UI for the Shopcart service. Built as a SPA (single-page app) using [Vue.js](https://vuejs.org/).

To run, just start the server using `python3 run.py` and visit `http://localhost:5000/`.

This adds basic CRUD functionality via the UI:
* Create, Update, Delete a Shopcart
* Add, Update, Delete Products in a Shopcart
* Prune Shopcarts

### UI Source Structure
* I'm using Vue.js (JS framework), Webpack, and NPM to create JavaScript components and compile them into a single js file.
* You can find the source code in `app\static\src`. The built/compiled files are in `app\static\dist` but should never be touched directly.
* To rebuild, make changes and then run `npm run build`from `\vagrant` on the VM.

### CODE REVIEW

**Comment by radhikamattoo** [comment]
I've never used Vue. before, but is it possible to add informative `id`s to important buttons, like `View`, `Prune`, `Create`, etc... so it's easier to find it in our integration tests? Like what the professor did [here](https://github.com/nyu-devops/lab-flask-bdd/blob/43d4d82d015ede6f566e7ef5785deb34758bf36b/features/steps/pet_steps.py#L54). 

So for example, the `View` button for a shopcart with id of `1` would have the attribute`id=view-btn-1`or something similar. That way it's a lot easier for us to find it, because right now we'd have to inspect the inner html of elements to find what we want.

**Comment by thoughts1053** [comment]
@radhikamattoo I agree with adding id's. I just didn't know what needed them. I can add them now or people can add them as they do the tests. Think you'll be comfortable rebuilding or just want me to add them? Anything else besides the ones you mentioned above?

**Comment by radhikamattoo** [comment]
@thoughts1053 I'm familiar with templating with hbs/handlebars but I don't really get Vue syntax, so I think it'd be faster for you to add them in. Other than that, everything looks great!

**Comment by thoughts1053** [comment]
@radhikamattoo I added id's to buttons and elements that we might have to query in the tests. Let me know if I missed anything.

File: app/static/src/App.vue [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<template>
+  <div id="app" class="container">
+    <h1 class="text-center mt-3">Shopcart Service</h1>
```
Comment (carpetri) [inline app/static/src/App.vue]: we should just name it "Shopcarts REST API Service" so that the Behave test pass. right?

Comment (thoughts1053) [reply app/static/src/App.vue]: We can change it or change the test. I'm not by a computer right now so if someone else wants to change it and rebuild via `npm run build`. Or we can merge and change it in a different PR.

**Pull request merged**